File: ProyectoMapa01/app/src/main/java/com/alejandro/proyectomapa01/MainActivity.java
package com.alejandro.proyectomapa01;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import android.content.Intent;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Intent intent = new Intent(this, MapsActivity.class);
        startActivity(intent);
    }
}

File: MagicAppV2/app/src/main/java/com/aalvarez/magicappv2/MainActivity.kt
package com.aalvarez.magicappv2

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.snackbar.Snackbar


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.book_list)

        val provider: ViewModelProvider = ViewModelProviders.of(this)
        val bookViewModel = provider.get(BookViewModel::class.java)

        val bookRecyclerView = findViewById<RecyclerView>(R.id.bookRecyclerView).apply {
            layoutManager = LinearLayoutManager(this@MainActivity)
            //layoutManager = GridLayoutManager(this@MainActivity,4, GridLayoutManager.HORIZONTAL,false)
            adapter = BookAdapter(bookViewModel.books, object : BookAdapter.OnItemClickListener{
                override fun onClicked(book: Book) {
                    val snackbar =
                        Snackbar.make(this@apply, book.titulo, Snackbar.LENGTH_LONG)
                    snackbar.show()
                }

            })
        }
    }
}